app/api/v1/auth/register/route.ts

```diff
@@ -0,0 +1,47 @@
+import { NextRequest, NextResponse } from 'next/server'
+import { hashSync } from 'bcryptjs'
+
+import knex from '@knex'
+
+export async function POST (request: NextRequest) {
+    const {
+        email,
+        username,
+        password,
+        confirmPassword,
+    } = await request.json()
+
+    try {
+        const user = await knex('users')
+            .where({ email })
+            .first()
+            .then((user: any) => user) // TODO: use knex interface builder
+        
+        if (user) {
+            return NextResponse.json({}, { status: 500 })
+        }
```

Comment:
Don't return a 500 error if it's not an internal server error. Return an Error 400. :)